### ISSUE TITLE
install_command from DistrubutionRequirement is broken

### ISSUE THREAD

**Issue by fneu** [coala-analyzer/coala#2700]
## TL;DR

different values for `install_command()` need confirmation that can never be given. The function also uses `platform.linux_distribution()`, which is deprecated in python 3.5 and will be removed in python 3.7.

## Explanation

So I was just trying out some stuff with cib on docker (ubuntu image), and stumbling on a bunch of stuff like this:

```
XMLBear is installing...  DONE!
  {'apt_get': 'libxml2'} (XMLBear's dependency) is installing...  FAILED!
Bears that failed installing/their dependencies failed installing:
XMLBear
```
I was wondering how system packages are installed anyway, and I think I found the reason for the error: There is an `install_command` created in  `coala/coalib/bears/requirements/DistributionRequirement.py` and cib uses `call_without_output = functools.partial(call, stdout=DEVNULL, stderr=DEVNULL)` from `coala/coalib/misc/Shell.py` to run that command and install the package.

Now, in this case that command would simply be `"apt-get install libxml2"`, right?

That command runs fine for me if I run it myself, but it asks for confirmation like so:

```
root@daa733272e15:/# apt-get install libxml2
Reading package lists... Done
Building dependency tree       
Reading state information... Done
libxml2 is already the newest version (2.9.3+dfsg1-1ubuntu0.1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
root@daa733272e15:/# apt-get install luarocks
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  autotools-dev libglib2.0-0 libglib2.0-data libidn11 libltdl-dev libltdl7 liblua5.1-0 liblua5.1-0-dev
  libreadline-dev libreadline6-dev libtinfo-dev libtool libtool-bin lua5.1 pkg-config shared-mime-info unzip wget
  xdg-user-dirs zip
Suggested packages:
  libtool-doc readline-doc autoconf automaken gfortran | fortran95-compiler gcj-jdk
The following NEW packages will be installed:
  autotools-dev libglib2.0-0 libglib2.0-data libidn11 libltdl-dev libltdl7 liblua5.1-0 liblua5.1-0-dev
  libreadline-dev libreadline6-dev libtinfo-dev libtool libtool-bin lua5.1 luarocks pkg-config shared-mime-info unzip
  wget xdg-user-dirs zip
0 upgraded, 21 newly installed, 0 to remove and 0 not upgraded.
Need to get 3537 kB of archives.
After this operation, 17.8 MB of additional disk space will be used.
Do you want to continue? [Y/n] 
```

That confirmation can never be given because input and output are suppressed 

## Solutions

So I guess the solution would be to **add something like`"--noconfirm"` to every command?**

Also, since `platform.linux_distribution()` is deprecated, doesn't work on all distributions (openSUSE, ...) and will be removed soon, maybe we can find another way to determine the installation method?
IMO it would be enough to just **check the availability of popular package managers** and use the first one available (zypper, dnf, yum, apt-get, apt, aptitude, pacman, portage, ...?).
That should also cover a much broader set of distributions.

**Comment by sils1297** [comment]
> MO it would be enough to just check the availability of popular package managers and use the first one > available (zypper, dnf, yum, apt-get, apt, aptitude, pacman, portage, ...?).
> That should also cover a much broader set of distributions.

agree, that can also help us getting rid of that map we have to maintain

**Comment by sils1297** [comment]
critical prio for making cib well usable, thanks @fneu for the investigation

**Comment by Adrianzatreanu** [comment]
this is the first part, with a lot of help from @SanketDG https://github.com/coala-analyzer/coala/pull/2679 and ill work on the second part once this is done :) Thanks, @fneu

**Comment by Makman2** [comment]
Indeed for `apt-get` we need to append `-y` (should work with many others too). So yes we need to maintain some little flag-database/dict ^^

We could also move the `install` command-line-parameter to this kind of database/dict, if other tools would possibly use a different one we could handle even that :+1:

**Comment by sils** [comment]
@fneu @Adrianzatreanu any news?
